### PR TITLE
delete @dotcom-tool-kit/options

### PR DESCRIPTION
**Pull request by apaleslimghost**
having the options stored in a singleton isn't very maintainable, and isn't compatible with the monorepo plugin's isolated config objects for workspace packages.

this gets rid of the options singleton and `options` package, replacing it with explicitly passing the options down from the `config` where they're needed by things that aren't tasks or hooks.

### CODE REVIEW

File: core/cli/src/tasks.ts [hunk 1]
```diff
@@ -97,7 +90,7 @@ export async function runTasksFromConfig(
     for (const task of tasks) {
       try {
         logger.info(styles.taskHeader(`running ${styles.task(task.id)} task`))
-        await task.run({ files, command, cwd: config.root })
+        await task.run({ files, command, cwd: config.root, config })
```
Comment (ivomurrell) [inline core/cli/src/tasks.ts:93]: **thought**: this is going to lead to future crimes i bet

**Review by ivomurrell — APPROVED** [review]

File: core/cli/src/tasks.ts (hunk 1 above)
Comment (apaleslimghost) [reply core/cli/src/tasks.ts:93]: yeah that was my worry too. if you can think of another way of achieving this i'm all ears!

File: lib/base/src/hook.ts [hunk 2]
```diff
@@ -5,14 +5,19 @@ import type { z } from 'zod'
 import type { Plugin } from '@dotcom-tool-kit/plugin'
 import { Conflict, isConflict } from '@dotcom-tool-kit/conflict'
 
+type Default<T, D> = T extends undefined ? D : T
```
[the remark this answers is not in the document]
Comment (apaleslimghost) [reply lib/base/src/hook.ts:8]: can do. i'm surprised `type-fest` doesn't have this type!

File: core/cli/src/tasks.ts (hunk 1 above)
Comment (ivomurrell) [reply core/cli/src/tasks.ts:93]: do you think we should `Object.freeze` the object before we pass it here to minimise the crime potential?

Comment (apaleslimghost) [reply core/cli/src/tasks.ts:93]: and make it `readonly` in the types

**Pull request merged**